integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -38,7 +39,8 @@ export class CdcJsService implements OnDestroy {
     protected auth: AuthService,
     protected zone: NgZone,
     protected userProfileFacade: UserProfileFacade,
-    @Inject(PLATFORM_ID) protected platform: any
+    @Inject(PLATFORM_ID) protected platform: any,
+    protected globalMessageService: GlobalMessageService
```

Comment:
To mitigate breaking change for customer, new contructor params should be optional (`?` in TS)

Reply:
Here's the link to the docs: 
https://sap.github.io/spartacus-docs/breaking-changes/#adding-new-constructor-dependencies-in-minor-versions

Reply:
Please discuss with Bill first the targeted release version, so you know which coding strategy to use (backward compatibiity vs. breaking change followed with migration docs 5_0.md and constructor update schematics)

Reply:
It is decided to go for 5.0 version, would be proceeding with documentation update.
thanks

Reply:
Therefore I think we should provide the migration docs (and possibly migration schematics) in this PR.

Reply:
Therefore I think we should provide the migration docs (and possibly migration schematics) in this PR.

Reply:
added migration docs